Add table github_package and github_package_version

ParthaI opened the pull request:
# Example query results
<details>
  <summary>Results</summary>
  
```
> select *  from github_package_version where organization = 'turbot' and package_name = 'steampipe/plugins/turbot/aws'
+----------------------+------------------------------+-----------+-------------------------------------------------------------------------+------------------------------------------------------------------------------------------------+---------------------------+--->
| login_id             | package_name                 | id        | digest                                                                  | html_url                                                                                       | tags                      | or>
+----------------------+------------------------------+-----------+-------------------------------------------------------------------------+------------------------------------------------------------------------------------------------+---------------------------+--->
| MDQ6VXNlcjQ3ODg3NTUy | steampipe/plugins/turbot/aws | 293319827 | sha256:91c6eeb0570b0229aba7c75628c7ada1015eeb727bef0ff44817a74744f4723a | https://github.com/orgs/turbot/packages/container/steampipe%2Fplugins%2Fturbot%2Faws/293319827 | ["11462847177","1.0.0"]   | tu>
| MDQ6VXNlcjQ3ODg3NTUy | steampipe/plugins/turbot/aws | 237235444 | sha256:69831633b99b174c86a968f2f60863f617a07b0f036d8d6ec12fcec9b35aa731 | https://github.com/orgs/turbot/packages/container/steampipe%2Fplugins%2Fturbot%2Faws/237235444 | ["9738491108","0.141.0"]  | tu>
| MDQ6VXNlcjQ3ODg3NTUy | steampipe/plugins/turbot/aws | 231280932 | sha256:217baa6355679212c7c13ccfb5994cbc6c36c1581a4475e08bedcc7f8576c5e0 | https://github.com/orgs/turbot/packages/container/steampipe%2Fplugins%2Fturbot%2Faws/231280932 | ["9554587442","0.139.0"]  | tu>
| MDQ6VXNlcjQ3ODg3NTUy | steampipe/plugins/turbot/aws | 214094141 | sha256:e37f0f833fc294ea5fce717364201456b38d89f76e505877bc138f19b7f83c25 | https://github.com/orgs/turbot/packages/container/steampipe%2Fplugins%2Fturbot%2Faws/214094141 | ["9016192446","0.138.0"]  | tu>
| MDQ6VXNlcjQ3ODg3NTUy | steampipe/plugins/turbot/aws | 203110114 | sha256:ed529dec9d3b7b36e7f49a5d78fe6953e81af7603dc511a33445d995ea3d9960 | https://github.com/orgs/turbot/packages/container/steampipe%2Fplugins%2Fturbot%2Faws/203110114 | ["8664910709","0.135.0"]  | tu>


> select * from github_package where organization = 'turbot'
+----------------------+---------+---------------------------------------------------------------------------------------------------------------+--------------+--------------+-------------+----------+-------------------------------------+---------------------------+-->
| login_id             | id      | name                                                                                                          | package_type | organization | owner_login | owner_id | owner_url                           | owner_html_url            | r>
+----------------------+---------+---------------------------------------------------------------------------------------------------------------+--------------+--------------+-------------+----------+-------------------------------------+---------------------------+-->
| MDQ6VXNlcjQ3ODg3NTUy | 6331189 | steampipe/plugins/turbot/docker                                                                               | container    | turbot       | turbot      | 38865304 | https://api.github.com/users/turbot | https://github.com/turbot | 3>
|                      |         |                                                                                                               |              |              |             |          |                                     |                           |  >
|                      |         |                                                                                                               |              |              |             |          |                                     |                           |  >
|                      |         |                                                                                                               |              |              |             |          |                                     |                           |  >
|                      |         |                                                                                                               |              |              |             |          |                                     |                           |  >
| MDQ6VXNlcjQ3ODg3NTUy | 1494498 | steampipe-postgres-fdw/run-on-arch-turbot-steampipe-postgres-fdw-build-linux-arm-binary-aarch64-ubuntu-latest | container    | turbot       | turbot      | 38865304 | https://api.github.com/users/turbot | https://github.com/turbot | 3>
|                      |         |                                                                                                               |              |              |             |          |                                     |                           |  >
|                      |         |                                                                                                               |              |              |             |          |                                     |                           |  >
|                      |         |                                                                                                               |              |              |             |          |                                     |                           |  >
|                      |         |                                                                                                               |              |              |             |          |                                     |                           |  >
| MDQ6VXNlcjQ3ODg3NTUy | 6331177 | steampipe/plugins/turbot/digitalocean                                                                         | container    | turbot       | turbot      | 38865304 | https://api.github.com/users/turbot | https://github.com/turbot | 3>

```
</details>

cbruno10 commented:
@ParthaI I think there were a few column name changes compared to the API property name changes, can you please list those in this PR? Thanks!

ParthaI commented:
@cbruno10, In this PR for the `github_package_version` table, I made the following changes intentionally:

- Added the column `package_name` instead of `name` to avoid conflicts with the `github_package` table, as it serves as the parent of `github_package_version`.
- Renamed the `name` column to `digest` since the value for the `name` attribute always contains the digest value (e.g., `sha256:91c6eeb0570b0229aba7c75628c7ada1015eeb727bef0ff44817a74744f4723a`)

I’d appreciate your thoughts on these changes.

Thanks!

ParthaI commented on github/table_github_package.go:
```diff
@@ -0,0 +1,136 @@
+			{Name: "updated_at", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("UpdatedAt").Transform(convertTimestamp), Description: "Timestamp when the package was last updated."},
+
+			// Package owner details
+			{Name: "owner_login", Type: proto.ColumnType_STRING, Transform: transform.FromField("Owner.Login"), Description: "Login name of the package owner."},
```
No problem. In general, for GitHub tables, we don’t always promote nested properties to the top level. I’ve removed the top-level fields from this table and kept only the `repository_full_name` column, as it’s commonly used across most tables and helps with joining data between them.

ParthaI commented on github/table_github_package_version.go:
```diff
@@ -0,0 +1,157 @@
+		Columns: commonColumns([]*plugin.Column{
+			{Name: "package_name", Type: proto.ColumnType_STRING, Description: "Name of the package."},
+			{Name: "id", Type: proto.ColumnType_INT, Description: "Unique ID of the package version.", Transform: transform.FromField("PackageVersion.ID")},
+			{Name: "digest", Type: proto.ColumnType_STRING, Description: "The digest (shasum) of the package version.", Transform: transform.FromField("PackageVersion.Name")},
```
I’ve kept the column name as `name` to match the API response. If you have a better suggestion, please let me know, I’m open to making changes. Thanks!
